pkg/server/token_minter.go

```diff
@@ -168,6 +168,11 @@ func (s *TokenMinterServer) processRequest(r *http.Request) *apiResponse {
 		return &apiResponse{http.StatusForbidden, fmt.Sprintf("no permissions available for scope %q in repository %q", request.Scope, claims.Repository), err}
 	}
 
+	// If there are no permissions in the request, use the set defined for the scope.
```

Comment:
So if i request nothing i get...?

Reply:
What is defined for that scope.

The goal being that the scopes defines the configuration of what you are allowed to do with it. 

You can reduce that by requesting less if you want but in most cases you will have the workflow match what's in config. This eliminates the need for that duplication.

Reply:
We log the request object (with or without permissions) 20 lines above this code. I don't think another log message that says "no requested permissions" or whatever will provide any value since I would probably check the token request first anyhow.